packages/fxa-auth-server/lib/routes/password.ts

```diff
@@ -644,74 +650,65 @@ module.exports = function (
         const { deviceId, flowId, flowBeginTime } = await request.app
           .metricsContext;
 
-        let passwordForgotToken;
-
-        return Promise.all([
+        await Promise.all([
           request.emitMetricsEvent('password.forgot.send_code.start'),
           customs.check(request, email, 'passwordForgotSendCode'),
-        ])
-
-          .then(db.accountRecord.bind(db, email))
-          .then((accountRecord) => {
-            if (
-              !emailsMatch(accountRecord.primaryEmail.normalizedEmail, email)
-            ) {
-              throw error.cannotResetPasswordWithSecondaryEmail();
-            }
-            // The token constructor sets createdAt from its argument.
-            // Clobber the timestamp to prevent prematurely expired tokens.
-            accountRecord.createdAt = undefined;
-            return db.createPasswordForgotToken(accountRecord);
-          })
-          .then((result) => {
-            passwordForgotToken = result;
-            return Promise.all([
-              request.stashMetricsContext(passwordForgotToken),
-              db.accountEmails(passwordForgotToken.uid),
-            ]);
-          })
-          .then(([_, emails]) => {
-            const geoData = request.app.geo;
-            const {
-              browser: uaBrowser,
-              browserVersion: uaBrowserVersion,
-              os: uaOS,
-              osVersion: uaOSVersion,
-              deviceType: uaDeviceType,
-            } = request.app.ua;
-
-            return mailer.sendRecoveryEmail(emails, passwordForgotToken, {
-              emailToHashWith: passwordForgotToken.email,
-              token: passwordForgotToken.data,
-              code: passwordForgotToken.passCode,
-              service: service,
-              redirectTo: request.payload.redirectTo,
-              resume: request.payload.resume,
-              acceptLanguage: request.app.acceptLanguage,
-              deviceId,
-              flowId,
-              flowBeginTime,
-              ip,
-              location: geoData.location,
-              timeZone: geoData.timeZone,
-              uaBrowser,
-              uaBrowserVersion,
-              uaOS,
-              uaOSVersion,
-              uaDeviceType,
-              uid: passwordForgotToken.uid,
-            });
-          })
-          .then(() =>
-            request.emitMetricsEvent('password.forgot.send_code.completed')
-          )
-          .then(() => glean.resetPassword.emailSent(request))
-          .then(() => ({
-            passwordForgotToken: passwordForgotToken.data,
-            ttl: passwordForgotToken.ttl(),
-            codeLength: passwordForgotToken.passCode.length,
-            tries: passwordForgotToken.tries,
-          }));
+        ]);
+        const accountRecord = await db.accountRecord(email);
```

Comment:
This seems much better, but flagging because the previous line `db.accountRecord.bind(db, email)` seems rather suspicous... Why would a bind have been needed?